src/Descriptor.cpp

```diff
@@ -111,3 +112,26 @@ bool Descriptor::write(std::string_view text) const {
     }
     return true;
 }
+
+namespace {
+
+// TODO duplicated with doorman! we should share implementation
+unsigned long djb2_hash(std::string_view str) {
+    unsigned long hash = 5381;
+    for (auto c : str)
+        hash = hash * 33 + c;
+    return hash;
+}
+
+// Returns the hostname, masked for privacy and with a hashcode of the full hostname. This can be used by admins to spot
+// users coming from the same IP.
+std::string get_masked_hostname(std::string_view hostname) {
+    return "{}*** [#{}]"_format(hostname.substr(0, 6), djb2_hash(hostname));
+}
+
+}
+
+void Descriptor::raw_full_hostname(std::string_view raw_full_hostname) {
+    raw_host_ = raw_full_hostname;
+    redacted_host_ = get_masked_hostname(raw_host_);
```

Comment:
masked/redacted,  should probably use the same word everywhere

Reply:
100%. changing to masked throughout